node_modules/oae-zencoder/tests/test-zencoder-api.js

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2016 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the 'License'); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an 'AS IS'
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+var _ = require('underscore');
+var assert = require('assert');
+var fs = require('fs');
+var util = require('util');
+
+var RestAPI = require('oae-rest');
+var TestsUtil = require('oae-tests/lib/util');
+
+var ZencoderAPI = require('oae-zencoder/lib/api');
+var ZencoderDAO = require('oae-zencoder/lib/internal/dao');
+
+describe('Zencoder API', function() {
+
+    var globalAdminRestContext = null;
+    var camAdminRestContext = null;
+    var globalAdminOnTenantRestContext = null;
+
+    /*!
+     * Set up rest contexts before each test
+     */
+    beforeEach(function(callback) {
+        globalAdminRestContext = TestsUtil.createGlobalAdminRestContext();
+        camAdminRestContext = TestsUtil.createTenantAdminRestContext(global.oaeTests.tenants.cam.host);
+
+        // Log in the admin so his cookie jar is set up appropriately
+        RestAPI.Admin.loginOnTenant(globalAdminRestContext, 'localhost', null, function(err, ctx) {
+            assert.ok(!err);
+            globalAdminOnTenantRestContext = ctx;
+
+            RestAPI.User.getMe(globalAdminOnTenantRestContext, function(err, user) {
+                assert.ok(!err);
+                assert.ok(!user.anon);
+                callback();
+            });
+        });
+    });
+
+    /*!
+     * @return a stream to a video file
+     */
+    var _getVideoStream = function() {
+        return fs.createReadStream(__dirname + '/data/video.mp4');
+    };
+
+    describe('#encodingCompleted', function() {
+        it('verify a valid notification is processed correctly', function(callback) {
+            TestsUtil.generateTestUsers(globalAdminOnTenantRestContext, 1, function(err, users) {
+                assert.ok(!err);
+                var user = _.values(users)[0];
+                var restCtx = user.restContext;
+                // Create the content item that we'll test with
+                RestAPI.Content.createFile(restCtx, 'test video', null, 'private', _getVideoStream, null, null, null, function(err, content) {
+                    assert.ok(!err);
+                    ZencoderDAO.saveContentRevisionId('10', content.id, content.latestRevisionId, function(err) {
```

Comment:
Should these kinds of tests go under the preview-processor then? The ones I've written were basically following the same pattern as the ones we currently have for mediacore [here](https://github.com/oaeproject/Hilary/blob/master/node_modules/oae-mediacore/tests/test-mediacore-processor.js#L143), with the exception of using nock to catch the requests instead of spinning up a server to respond in lieu of Zencoder. It seems like tests for the actual preview processing process should go into that module instead of replicating the setup here.